src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRMultiConnectionInfoBinding/MultiConnectionInfoAsyncConverter.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.SignalR;
+using Microsoft.Azure.SignalR.Management;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal class MultiConnectionInfoAsyncConverter : IAsyncConverter<SignalRMultiConnectionInfoAttribute, Dictionary<ServiceEndpoint, SignalRConnectionInfo>>
```

Comment:
@vicancy doesn't like abbr. in naming

Reply:
We already have `MultiEndpointServiceConnectionContainer` in SDK? If we don't use abbr. , then the name will be super long. I don't think this is a good experience for users and us.

Reply:
maybe some `ConnectionInfoMap` instead of `Multi...`

Reply:
I found using a endpoint object as the map key is not convenient to convert the map to JSON. Intend to change the map to a array like this:
```json
[
    {
          "endpoints":{},
          "connectionInfo":{}
    },
]
```
Giving this, we can still use `MultiConnectionInfo`?

Reply:
> endpoint object as the map key is not convenient to convert the map to JSON

where is inconvenient?

Reply:
A potential issue for using array is array is not extensible for adding more metadata properties, so generally we would prevent using array as the top level type, is this JSON array the top type?

Reply:
what about `ConnectionInfos` or `NegotiationResponses`?

Reply:
How about `NegotiateContext`

Reply:
what about reuse the standard one https://github.com/dotnet/aspnetcore/blob/main/src/SignalR/common/Http.Connections.Common/src/NegotiationResponse.cs , signalr users should have knowledge about it @vicancy

Reply:
JSON can not represent a map whose keys are object.

Reply:
about the naming, since we already has `SignalRConnectionInfo`, and now we decide to returns a list, how about naming the return type as `SignalRConnectionInfoList` and `[SignalRConnectionInfoList]`? @JialinXin

Reply:
> I found using a endpoint object as the map key is not convenient to convert the map to JSON
how about using endpoint name?
Consider "Resiliency and disaster recovery", they already has the key as `primary` and `secondary`, they try to get the primary first, if its offline, then to the secondary. So a dictionary is more convenient.

Reply:
What is the type of es?

Reply:
"they already has the key as primary and secondary, they try to get the primary first, if its offline, then to the secondary. So a dictionary is more convenient. " 
I am not sure I understand your meaning. Do you mean that the user already has a list of names of primary endpoints , and they will loop the dictionary by the names? But I don't think user would maintain a list of names of endpoints. What's more, a dictionary is not convenient for users to get a random endpoint.

Reply:
I just find that since ES6, we can loop over object by Object.Keys(), so we could consider it.

Reply:
yeah.

Reply:
But SDK just return a list to user. Using a list is more consistent with SDK.